### DISCUSSION TITLE
Private Cloud support

### DISCUSSION THREAD

**Discussion by moonlight16** [skypilot-org/skypilot#1935] [General]
Does this technology work on private clouds?  or local/virtual environments?

I want to be able to try out this technology without actually paying for AWS, Azure, ...

Or any suggestions just to play around in development mode or something like that?

**Comment by romilbhardwaj** [comment]
Hi @moonlight16! We currently have [experimental support](https://skypilot.readthedocs.io/en/latest/reference/local/index.html) for running SkyPilot on your onprem cluster. This would allow you to run tasks on an onprem cluster you have ssh access to. Note that it's experimental code, and there are lots of rough edges there.

In the medium-term, we're working on adding support for Kubernetes. This would allow running on private clouds/clusters that run Kubernetes, and we hope to have a robust offering for this. Once we have Kubernetes support, we can also support a development mode running on your machine which would allow you to run SkyPilot tasks locally. Stay tuned!
